src/main/java/com/learning/pages/swaglabs/SwagLabsInventoryPage.java

```diff
@@ -38,20 +32,16 @@ public void addItemToCart(String itemName) {
   }
 
   public int countItemsInCart() {
-      return Integer.parseInt(cart.getText());
+    return Integer.parseInt(cart.getText());
   }
 
-  @Override
-  public boolean waitForPageToLoad() {
-    WebDriverWait wait = new WebDriverWait(driver, 10);
-    return wait.until(ExpectedConditions
-        .visibilityOfAllElements(offeringCards))
-        .stream()
-        .allMatch(Objects::nonNull);
+  String getURL() {
+    return swagLabsInventoryURL;
   }
 
-  String getURL() {
-    return propertiesReader.getProperty(INVENTORY_PAGE_URL_PROPERTY);
+  @Override
+  public boolean isLoaded() {
+    return browser.waitForElementsToLoad(offeringCards.toArray(WebElement[]::new));
```

Comment:
that transformation to array should not be needed, `waitForElementsToLoad` method should be used like this `waitForElementsToLoad(elementOne, elementTwo, elementThree)`, there should be no need to wrap it with collection.

Reply:
But `offeringCards` already are `List<WebElement>` so other way to pass it is to get single elements from list.

Reply:
ahh.. right... I didn't notice :/